### DISCUSSION TITLE
Find best way to do integration testing of the Relay

### DISCUSSION THREAD

**Discussion by MuhammedIrfan** [icon-project/icon-bridge#626] [Q&A]
## Summary
At present there is no integration test being done on the relay, There where couple of bugs which had been identified which could have been caught earlier if there was integration tests in place. 

## Challenge
As the relay have minimal input and output, this make automation testing of the relay difficult, At present relay requires inputs as config, websocket and json rpc, and output via json rpc and logs, Hence it requires to create a mock json rpc server to provide test data. The test outcome need to be captured using json rpc server or logs which need to be parsed.

**Comment by CyrusVorwald** [comment]
@andrii-kl please take a look at this

**Reply by andrii-kl** [reply]
I will think it over about this point. 
But in any case, we can concentrate on the unit tests now. BMR has quite low unit test coverage level.

**Reply by CyrusVorwald** [reply]
Sounds good
